### PR TITLE
Add multisampled texture type

### LINKED ISSUE #990: Sampling from MSAA textures
I realized that the MSAA textures are in a weird position: they are "sampled texture" binding types on both the API and shaders sides today, but they can't really be sampled. In GLSL, they can also be "texel fetched", which corresponds to a `texture_load` in WGSL today, which the grammar only allows to happen on storage textures (with a TODO related to sampled and multi-sampled textures).

## Proposal
Here is a way we could resolve this:
  1. have a separate "multisampled-texture" binding type, Edit: and remove the "multisampled" flag
  2. have a separate corresponding set of types on WGSL. That would mean `texture_sample` is no longer applicable to it, which is what we need.
  3. have `texture_load` functions overloaded to work on multisampled textures. They need an extra "sample ID" integer parameter for these overloads. They would also need to be overloaded for regular sampled textures.

### PR DESCRIPTION
**Pull request by kvark**
Closes #990

### CODE REVIEW

**Comment by jdashg** [comment]
Approved to land after rebase.

**Comment by kvark** [comment]
Rebased now

File: wgsl/index.bs [hunk 1]
```diff
@@ -2977,11 +2990,15 @@ TODO: deduplicate these tables
 `vec4<type> textureLoad(texture_ro, i32 coords, i32 level_of_detail)`
 `vec4<type> textureLoad(texture_ro, vec2<i32> coords, i32 level_of_detail)`
 `vec4<type> textureLoad(texture_ro, vec3<i32> coords, i32 level_of_detail)`
-  %32 = OpImageFetch %v4float %texture_ro %coords Lod %level_of_detail
+  %32 = OpImageLoad %v4float %texture_ro %coords Lod %level_of_detail
```
Comment (ponitka) [inline wgsl/index.bs:2993]: Shouldn't this be OpImageRead?

Comment (kvark) [reply wgsl/index.bs:2993]: Oh yes, that's a mistake. Would you want to make a PR fixing this?

Comment (ponitka) [reply wgsl/index.bs:2993]: I made one.

**Pull request merged**